arch/riscv/core/asm_macros.inc

```diff
@@ -32,3 +32,25 @@
 	.endm
 
 #endif
+
+#ifdef CONFIG_CPU_HAS_FPU_DOUBLE_PRECISION
```

Comment:
If  another extension-related ifdef is being added, maybe it would be a good idea to name it as described [here](https://github.com/zephyrproject-rtos/zephyr/pull/43557/files#r822678126), i.e. `CONFIG_RISCV_ISA_EXT_D`?

Reply:
Also, many empty lines could and probably should be removed from this file.